2 - RTB/Documentazione esterna/Analisi dei Requisiti/Analisi dei Requisiti.typ

```diff
@@ -0,0 +1,341 @@
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+l'ambiente di lavoro deve essere stato configurato con successo.
+
+$bold("Postcondizioni: ")$
+nell'ambiente di lavoro è stato aggiunto un nuovo scaffale.
+
+$bold("Scenario: ")$
+- l'utente seleziona l'aggiunta di uno scaffale;
+- l'utente inserisce i dati necessari alla creazione dello scaffale;
+- l'utente posiziona lo scaffale in una posizione valida nell'ambiente di lavoro.
+
+
+
+== UC-5 Modifica scaffale
+$bold("Descrizione: ")$ 
+modifica dei valori di uno scaffale già esistente.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+nell'ambiente deve essere posizionato almeno uno scaffale.
+
+$bold("Postcondizioni: ")$
+i valori di uno scaffale scelto sono stati modificati come indicato.
+
+$bold("Scenario: ")$
+- l'utente seleziona uno scaffale nell'ambiente di lavoro;
+- l'utente seleziona il comando per la modifica dello scaffale;
+- l'utente inserisce i nuovi valori.
+
+$bold("Estensioni: ")$
+UC-5.1 Visualizzazione errore inserimento valori errati per la modifica di uno scaffale.
+
+
+
+=== UC-5.1 Visualizzazione errore inserimento valori errati per la modifica di uno scaffale 
+$bold("Descrizione: ")$ 
+i dati inseriti per la modifica di uno scaffale sono errati e non possono essere accettati.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+-l'attività di modifica di uno scaffale deve essere stata attivata;
+-l'inserimento dei valori per la modifica deve essere avvenuto.
+
+$bold("Postcondizioni: ")$
+all'utente viene notificato l'errore relativo ai dati errati.
+
+$bold("Scenario: ")$
+- l'utente ha avviato la procedura di modifica di uno scaffale esistente;
+- l'utente ha immesso dati non validi o che comporterebbero incongruenze con i bin o altri elementi dell'ambiente.
+
+
+== UC-6 Eliminazione scaffale
+$bold("Descrizione: ")$ 
+lo scaffale selezionato presente nell'ambiente viene eliminato.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+- nell'ambiente deve essere posizionato almeno uno scaffale;
+- la modalità di modifica dell'ambiente deve essere attiva;
+- lo scaffale da eliminare deve contenere solo bin vuoti.
+
+$bold("Postcondizioni: ")$
+- lo scaffale selezionato viene rimosso dall'ambiente;
+- vengono rimossi i bin in esso contenuti.
+
+$bold("Scenario: ")$
+- l'utente seleziona uno scaffale nell'ambiente;
+- l'utente seleziona il comando per la rimozione dello scaffale;
+- l'utente conferma l'operazione da una finestra di conferma.
+
+$bold("Estensioni: ")$
+UC-6.1 Visualizzazione errore scaffale da eliminare non vuoto.
+
+
+=== UC-6.1 Visualizzazione errore scaffale da eliminare non vuoto
+$bold("Descrizione: ")$ 
+è stata richiesta l'eliminazione di uno scaffale contenente almeno un bin non vuoto.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+- l'attività di eliminazione di uno scaffale deve essere stata attivata;
```

Comment:
"l'attività di eliminazione di uno scaffale deve essere stata attivata;"? ... direi più che come pre condizioni potrebbero essere:
- configurazione di sistema correttamente caricata
- presenza di almeno uno scaffale con bin non vuoti
- la procedura di eliminazione interessa uno scaffale con almeno un bin non vuoto

Reply:
se per "configurazione di sistema correttamente caricata" intendi la configurazione dell'ambiente di lavoro allora è implicito dal momento che "l'attività di eliminazione di uno scaffale deve essere stata attivata" implica che sia stata attivata l'uc 6 e questa ha come pre condizioni " nell'ambiente deve essere posizionato almeno uno scaffale", a sua volta tale pre condizione implica l'attivazione dell' uc 4 che come pre condizione ha "- l'ambiente di lavoro deve essere stato configurato con successo."

"presenza di almeno uno scaffale con bin non vuoti" e "la procedura di eliminazione interessa uno scaffale con almeno un bin non vuoto" sono esplicitati in " tale scaffale contiene almeno un bin non vuoto."

Reply:
Allora cambierei semplicemente il “tale” in un “Lo scaffale interessato contiene almeno un bin non vuoto”.